From aaacd2e91006e562de4cc8994080947cf2724e7b Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Tue, 4 Jun 2019 20:03:35 +0300
Subject: [PATCH 901/909] lvds: geosemi: put imager to autodetect tail

commit a0b6b13ecaa44559b02574814e941d5fbbf6589c from
https://github.com/CogentEmbedded/meta-rcar.git

Put Geosemi cameras to tail since their produce ID is
just an i2c address

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/i2c/soc_camera/ov106xx.c | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/drivers/media/i2c/soc_camera/ov106xx.c b/drivers/media/i2c/soc_camera/ov106xx.c
index 442e54a358b6..36aa88a807b7 100644
--- a/drivers/media/i2c/soc_camera/ov106xx.c
+++ b/drivers/media/i2c/soc_camera/ov106xx.c
@@ -129,18 +129,6 @@ static int ov106xx_probe(struct i2c_client *client,
 		goto out;
 	}
 
-	ret = gw4200_probe(client, did);
-	if (!ret) {
-		chip_id = ID_GW4200_AR014X;
-		goto out;
-	}
-
-	ret = gw5200_probe(client, did);
-	if (!ret) {
-		chip_id = ID_GW5200_IMX390;
-		goto out;
-	}
-
 	ret = ov2775_probe(client, did);
 	if (!ret) {
 		chip_id = ID_OV2775;
@@ -171,6 +159,18 @@ static int ov106xx_probe(struct i2c_client *client,
 		goto out;
 	}
 
+	ret = gw4200_probe(client, did);
+	if (!ret) {
+		chip_id = ID_GW4200_AR014X;
+		goto out;
+	}
+
+	ret = gw5200_probe(client, did);
+	if (!ret) {
+		chip_id = ID_GW5200_IMX390;
+		goto out;
+	}
+
 	v4l_err(client, "failed to probe @ 0x%02x (%s)\n",
 			client->addr, client->adapter->name);
 out:
-- 
2.17.1

